compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEInsertCores.cpp

```diff
@@ -122,6 +122,27 @@ LogicalResult insertCoreOps(mlir::ModuleOp moduleOp) {
         }
         rewriter.setInsertionPoint(endOp);
         rewriter.moveOpBefore(currOp, endOp);
+      } else if (auto callOp = dyn_cast<func::CallOp>(op)) {
+        // Fetch name of the ukernel function to look up its declaration in the
+        // Symbol table.
+        StringRef fnName = callOp.getCallee();
+        auto fnDecl = dyn_cast_or_null<func::FuncOp>(
+            SymbolTable::lookupSymbolIn(moduleOp, fnName));
+        assert(fnDecl && "expected function declaration");
+        assert(fnDecl->hasAttr("link_with") &&
+               "expected 'link_with' construct for the function declaration");
+        // From the declaration of the function, we extract the value of
+        // attribute "link_with" and attach it to amdaie.core op.
+        // TODO(avarma): What to do when more than one func.call has different
```

Comment:
This seems like like a pretty important feature we might need in the future and we could support it with a `Variadic`? But up to you on whether you want to implement it now.

Reply:
Absolutely! In fact, this would need changes to be thought of in the `lower-to-ukernel` + the actual ukernel `.o` formation.

If we club all possible ukernels within a single `.o` then the above issue should never be hit.
Also for single dispatches too this doesn't become an issue.

So, will table this for later when we gradually mature this and as and when the need arises.